airflow/utils/state.py

```diff
@@ -111,3 +111,7 @@ def unfinished(cls):
             cls.RUNNING,
             cls.UP_FOR_RETRY
         ]
+
+    @classmethod
+    def unsuccessful(cls):
```

Comment:
This is necessary since we need to know whether task_instance.state is in unsuccessful so that we can judge whether we need to call self.task_instance.handle_failure(e) again.